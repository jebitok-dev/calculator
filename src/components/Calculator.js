import React from 'react';
import App from './App';

const Calculator = () => (
  <div className="centered">
    <h2>Lets do some Math!</h2>
    <div>
      <App />
    </div>
  </div>
);

export default Calculator;
