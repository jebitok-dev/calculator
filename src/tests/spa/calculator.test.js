import renderer from 'react-test-renderer';
import React from 'react';
import Calculator from '../../components/calculator';

it('should match Calculator snapshot', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
