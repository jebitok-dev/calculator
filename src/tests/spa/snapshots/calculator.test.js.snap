// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render calculator page`] = `
<div className="centered">
    <h2>Lets do some Math!</h2>
    <div>
        <div className="is-flex calculator">
            <div className="is-display" data-testid="display-panel">
                <h2 className="display-heading">0</h2>
            </div>
            <div data-testid="button-panel">
                <div className="calculator-row">
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">AC</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">+/-</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">%</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">/</button>
                </div>
                <div className="calculator-row">
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">7</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">8</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">9</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">X</button>
                </div>
                <div className="calculator-row">
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">4</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">5</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">6</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">-</button>
                </div>
                <div className="calculator-row">
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">1</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">2</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">3</button>
                    <button type="button" className="button" onClick={[Function]}
 data-testid="button">+</button>
                </div>
                <div className="calculator-row">
            <button type="button" className="button" onClick={[Function]}
 data-testid="button">0</button>
            <button type="button" className="button" onClick={[Function]}
 data-testid="button">.</button>
            <button type="button" className="button" onClick={[Function]}
 data-testid="button">=</button>
                </div>
            </div>
        </div>
    </div>
</div>

`;